syne_tune/optimizer/schedulers/multiobjective/multi_surrogate_multi_objective_searcher.py

```diff
@@ -0,0 +1,155 @@
+    :param num_initial_candidates: Number of candidates sampled for scoring with
+        acquisition function.
+    :param num_initial_random_choices: Number of randomly chosen candidates before
+        surrogate model is used.
+    :param allow_duplicates: If ``True``, allow for the same candidate to be
+        selected more than once.
+    :param restrict_configurations: If given, the searcher only suggests
+        configurations from this list. If ``allow_duplicates == False``,
+        entries are popped off this list once suggested.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        estimators: Dict[str, Estimator],
+        mode: List[str] = None,
+        points_to_evaluate: Optional[List[Dict[str, Any]]] = None,
+        scoring_class_and_args: Optional[ScoringClassAndArgs] = None,
+        num_initial_candidates: int = DEFAULT_NUM_INITIAL_CANDIDATES,
+        num_initial_random_choices: int = DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+        allow_duplicates: bool = False,
+        restrict_configurations: Optional[List[Dict[str, Any]]] = None,
+        clone_from_state: bool = False,
+        **kwargs,
+    ):
+        if mode is None:
+            mode = ["min" for item in metric]
+
+        super().__init__(
+            config_space,
+            metric=metric,
+            mode=mode,
+            points_to_evaluate=points_to_evaluate,
+            random_seed_generator=kwargs.get("random_seed_generator"),
+            random_seed=kwargs.get("random_seed"),
+        )
+        self.estimator = estimators
+        if scoring_class_and_args is None:
+            scoring_class_and_args = MultiObjectiveLCBRandomLinearScalarization
+
+        if not clone_from_state:
+            hp_ranges = make_hyperparameter_ranges(self.config_space)
+            self._create_internal(
+                hp_ranges=hp_ranges,
+                estimator=self.estimator,
+                acquisition_class=scoring_class_and_args,
+                num_initial_candidates=num_initial_candidates,
+                num_initial_random_choices=num_initial_random_choices,
+                initial_scoring="acq_func",
+                skip_local_optimization={name: True for name in self.estimator.keys()},
+                allow_duplicates=allow_duplicates,
+                restrict_configurations=restrict_configurations,
+                **kwargs,
+            )
+        else:
+            # Internal constructor, bypassing the factory
+            # Note: Members which are part of the mutable state, will be
+            # overwritten in ``_restore_from_state``
+            self._create_internal(**kwargs.copy())
+
+    def _update(self, trial_id: str, config: Dict[str, Any], result: Dict[str, Any]):
+        # Gather metrics
+        metrics = {name: result[name] for name in self._metric}
+
+        # Cost value only dealt with here if ``resource_attr`` not given
```

Comment:
Is your searcher using `self._cost_attr`? If not, you can remove this here. This is just a way for me in `GPFIFOSearcher` to allow for certain cost-aware variants in subclasses. No need to copy that here.

I think that even if "cost" is one of your metrics, you just want to not single it out here.